xbmc/epg/EpgContainer.cpp

```diff
@@ -65,7 +65,10 @@ CEpgContainer::CEpgContainer(void) :
 
 CEpgContainer::~CEpgContainer(void)
 {
-  Unload();
+  // Don't call Unload here, it'll try
+  // to notify observers which may be in
+  // different states of teardown
+  Stop();
```

Comment:
Imo, reaching dtor does not necessarily mean Kodi "shutdown". Thus I'm not sure this is okay.

Unload() does much more than Stop().